src/coreclr/ToolBox/superpmi/superpmi/icorjitinfo.cpp

```diff
@@ -1609,7 +1609,21 @@ void MyICJI::allocMem(ULONG              hotCodeSize,   /* IN */
     jitInstance->mc->cr->AddCall("allocMem");
 
     // TODO-Cleanup: Could hot block size be ever 0?
-    *hotCodeBlock = jitInstance->mc->cr->allocateMemory(hotCodeSize);
+    size_t codeAlignment      = sizeof(void*);
```

Comment:
I'm worried there is a bug here. I think one side-effect of your loop alignment change is that we'll generate different code based on the actual alignment of the addresses of the instructions. You've changed SuperPMI here to align the code properly based on flags. But you don't save this increased sized and then use it for certain replays. In particular, `CompileResult::recAllocMem` saves what has been allocated, `CompileResult::recAllocMemCapture` stores it into the MC. But it stores it without the alignment info or size. Normally, in our processing, we delete the CompileResult, so you might think it's not a problem. However, asm diffs uses NearDiffer to generate diffs, and `NearDiffer::compare` uses `repAllocMem` to "replay" the allocation, and that has no provision for alignment. I'm worried what will happen is you'll get spurious diffs due to generated code that is changed based on alignment, but alignment wasn't saved/replayed properly.

I think you need to figure out how to save/restore the alignment and aligned-up allocation sizes.